### PR TITLE
Add a notification daemon

### PR DESCRIPTION
**Pull request by balsoft**
This PR adds noted, a notification daemon for termNote. It uses libnotify, so libnotify should probably be added as a dependency in README. This builds and works when using nix.

### CODE REVIEW

**Comment by balsoft** [comment]
Whoops, I have messed up a bit. This needs some more files.

File: src/noted.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
+#include <argp.h>
+#include <iostream>
+#include <libnotify/notify.h>
+#include "../include/note.hpp"
+#include <stdlib.h>
```
Comment (ghost) [inline src/noted.cpp]: there is already <cstdlib> included in note.hpp

Comment (ghost) [reply src/noted.cpp]: cstdlib*

File: src/noted.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
+#include <argp.h>
+#include <iostream>
```
Comment (ghost) [inline src/noted.cpp:2]: There is already iostream and argp in note.hpp included

File: src/noted.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/inotify.h>
+#include <limits.h>
+#include <chrono>
+#include <thread>
+#include <ctime>
+#define NOTE_DUE "One of your notes just reached its due time!"
```
Comment (ghost) [inline src/noted.cpp]: Why not use a static const?

**Review by ghost — CHANGES_REQUESTED** [review]
Please remove the includes that are already included in the notes.hpp, also why #DEFINE instead of static const std::string?

File: src/noted.cpp (hunk 2 above)
Comment (balsoft) [reply src/noted.cpp:2]: I don't think argp is included. Removed all unneeded includes.

File: src/noted.cpp (hunk 1 above)
Comment (balsoft) [reply src/noted.cpp]: Fixed, thanks.

File: src/noted.cpp (hunk 3 above)
Comment (balsoft) [reply src/noted.cpp]: This is only used once throughout the code, so I see practically no difference in binary size/mem usage/speed/etc. libnotify is a C library, so it requires a `char*`. Anyways, changed this to `const char* NOTE_DUE = ...`

**Comment by balsoft** [comment]
Also, why are you using tabs and not spaces? If this matters to you, then I'll change my emacs config to use tabs on this project, otherwise I would very much prefer two-spaces (or four-spaces) to tabs.

**Comment by ghost** [comment]
Because i used vscode instead of my usual vim config... i will go and change the Tabs to 4 spaces.

**Comment by balsoft** [comment]
> Because i used vscode instead of my usual vim config... i will go and change the Tabs to 4 spaces.

Ok, may I untabify everything then?

**Comment by ghost** [comment]
I woul'd have done it myself, but if you want, go and PR it

**Pull request merged**